plugins/services/src/js/utils/ServiceConfigUtil.js

```diff
@@ -19,6 +19,25 @@ const ServiceConfigUtil = {
     const hostname = HostUtil.stringToHostname(id);
 
     return `${hostname}${Networking.L4LB_ADDRESS}:${port}`;
+  },
+
+  buildHostNameFromVipLabel(label, port) {
+    const [ipOrName, labelPort] = label.split(':');
+
+    // Sometimes we don't know the port yet
+    port = port || '<assigned port>';
+
+    // 0 means randomly assigned port
+    port = labelPort === '0' ? port : labelPort;
```

Comment:
Do we need to consider the case where `labelPort` is undefined if `label` does not contain `:` ?